package/MDAnalysis/guesser/base.py

```diff
@@ -0,0 +1,199 @@
+import copy
+
+logger = logging.getLogger("MDAnalysis.guesser.base")
+
+
+class _GuesserMeta(type):
+    """Internal: guesser classes registration
+
+    When classes which inherit from GuesserBase are *defined*
+    this metaclass makes it known to MDAnalysis.  'context'
+    attribute  are read:
+     - `context` defines the context of the guesser class for example:
+       forcefield specific context  as MartiniGuesser
+       and file specific context as PDBGuesser.
+
+    Eg::
+
+      class FooGuesser(GuesserBase):
+          format = 'foo'
+
+    .. versionadded:: 2.7.0
+    """
+    def __init__(cls, name, bases, classdict):
+        type.__init__(type, name, bases, classdict)
+
+        _GUESSERS[classdict['context'].upper()] = cls
+
+
+class GuesserBase(metaclass=_GuesserMeta):
+    """Base class for context-specific guessers to inherit from
+
+    Parameters
+    ----------
+    universe : Universe, optional
+        Supply a Universe to the Guesser. This then become the source of atom
+        attributes to be used in guessing processes. (this is relevant to how
+        the universe's guess_topologyAttributes API works.
+        See :ref:`guess_TopologyAttributes <guess_TopologyAttributes>`).
+    **kwargs: to pass additional data to the guesser that can be used with
+              different methos.
+
+    .. versionadded:: 2.7.0
+
+    """
+    context = 'base'
+    _guesser_methods: Dict = {}
+
+    def __init__(self, universe=None, **kwargs):
+        self._universe = universe
+        self._kwargs = kwargs
+
+    def update_kwargs(self, **kwargs):
+        self._kwargs.update(kwargs)
+
+    def copy(self):
+        """Return a copy of this Guesser"""
+        kwargs = copy.deepcopy(self._kwargs)
+        new = self.__class__(universe=None, **kwargs)
+        return new
+    
+    def are_guessable(self, attrs_to_guess):
+        """check if the passed atrributes can be guessed by the guesser class
+
+        Parameters
+        ----------
+        guess: list
+            Atrributes to be guessed then added to the universe
+
+        Returns
+        -------
+        Boolean value
+        """
+        if attrs_to_guess:
+            for a in attrs_to_guess:
+                if a.lower() not in self._guesser_methods:
+                    return False
+            return True
+
+    def guess_attr(self, attr_to_guess, force_guess=False):
+        """map the attribute to be guessed with the apporpiate guessing method
+
+        Parameters
+        ----------
+        attr_to_guess: list
+            an atrribute to be guessed then to be added to the universe
+        force_guess: boolean
+            To indicate wether to only partialy guess the empty values of the
+            attribute or to overwrite all existing values by guessed one
+
+        Returns
+        -------
+        list of guessed values
+
+        Raises
+        ------
+        ValueError
+            Failed to find or guess parent attribute
```

Comment:
> Could you clarify or link to an explanation of what a parent attribute would be in this case?
This may be an outdated documentation of a previous implementation... I'll remove it